src/ad-sizes.ts

```diff
@@ -0,0 +1,87 @@
+export type AdSizeString = 'fluid' | `${number},${number}`;
+
+export type AdSize = Readonly<{
+	width: number;
+	height: number;
+	toString: () => AdSizeString;
+}>;
+
+export type SizeKeys =
+	| 'billboard'
+	| 'leaderboard'
+	| 'mpu'
+	| 'halfPage'
+	| 'portrait'
+	| 'skyscraper'
+	| 'mobilesticky'
+	| 'fluid'
+	| 'outOfPage'
+	| 'googleCard'
+	| 'video'
+	| 'outstreamDesktop'
+	| 'outstreamGoogleDesktop'
+	| 'outstreamMobile'
+	| 'merchandisingHighAdFeature'
+	| 'merchandisingHigh'
+	| 'merchandising'
+	| 'inlineMerchandising'
+	| 'fabric'
+	| 'empty'
+	| '970x250'
+	| '728x90'
+	| '300x250'
+	| '300x600'
+	| '300x1050'
+	| '160x600';
+
+const getAdSize = (width: number, height: number): AdSize => {
+	const toString = (): AdSizeString =>
+		width === 0 && height === 0 ? 'fluid' : `${width},${height}`;
+
+	return Object.freeze({
```

Comment:
Not related to the PR - I know we've replicated this across from Frontend but I'm curious to understand why we create a frozen object with a `toString`?

Just thinking ahead to futureland where we try and simplify the codebase, could this just be a string - in which case we don't need `getAdSize` and can just string template the sizes?

Something we can discuss with the team.

Reply:
Good point - I wonder whether we still need to maintain the readonly object (we wouldn't need it in DCR) but are the `.width` and `.height` properties used in frontend? Will note this as a future change we should make.